dlt/dataset/dataset.py

```diff
@@ -170,6 +175,46 @@ def is_same_physical_destination(self, other: dlt.Dataset) -> bool:
         """
         return is_same_physical_destination(self, other)
 
+    # TODO explain users can inspect `_dlt_loads` table to differentiate data originating
+    # from `pipeline.run()` or `dataset.write()`
+    @contextmanager
+    def write_pipeline(self) -> Generator[dlt.Pipeline, None, None]:
+        """Get the internal pipeline used by `Dataset.write()`.
+
+        Passing a `pipelines_dir` allows you to set a
+        """
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            pipeline = _get_internal_pipeline(
+                dataset_name=self.dataset_name, destination=self._destination, pipelines_dir=tmp_dir
+            )
+            yield pipeline
+
+    def write(
+        self,
+        data: TDataItems,
+        *,
+        table_name: str,
+        write_disposition: TWriteDisposition = "append",
+    ) -> LoadInfo:
+        """Write `data` to the specified table.
+
+        This method uses a full-on `dlt.Pipeline` internally. You can retrieve this pipeline
+        using `Dataset.get_write_pipeline()` for complete flexibility.
+        """
+        with self.write_pipeline() as internal_pipeline:
+            # TODO should we try/except this run to gracefully handle failed writes?
+            info = internal_pipeline.run(
+                data,
+                dataset_name=self.dataset_name,
+                table_name=table_name,
+                schema=self.schema,
+                write_disposition=write_disposition,
+            )
+
+        # maybe update the dataset schema
+        self._schema = internal_pipeline.default_schema
```

Comment:
For sure - you want to see new tables right away, I'd extract this to a method and verify if we need to purge any cached objects (like sql_client which also holds schema)